src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -139,6 +139,15 @@ public async Task<MetadataAsSourceFile> GetGeneratedFileAsync(Project project, I
                         }
                         catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
                         {
+                            // Insert the exception message at the top of the fallback Metadata As Source document to
+                            // help with diagnosing failures. When the generated source is added to the document, this
+                            // comment will end up at the bottom.
+                            var failureText = e.ToString();
+                            var failureTextAsComment = "// " + failureText.Replace("\n", "\n// ") + "\n";
```

Comment:
Don't these need to be \r\n if we're generating into a CRLF file? Is this going to result in inconsistent line endings?

Reply:
(can we add a test either way? Since the test for this more or less duplicates this line, it seems we don't really have a test for this standalone)

Reply:
Updated to use system line endings

Reply:
The code is run through the formatter here, so the specific line endings don't matter:

https://github.com/dotnet/roslyn/blob/052a7c89739dc73f1bd26910ce9dc2ae1c573d5e/src/Features/Core/Portable/MetadataAsSource/AbstractMetadataAsSourceService.cs#L56-L57